### ISSUE TITLE
Receiving 'Could not extract and write stream to file system.' error when creating a new project in MM 5.0.0 

### ISSUE THREAD

**Issue by lacollier** [joeferraro/MavensMate-SublimeText#683]
I filled out my project details, and clicked on 'Create MavensMate Project', then I received the error.  I changed my log settings to 'SILLY', and here is some relevant lines:

```
2015-10-01T18:32:15.122Z - debug: writing stream
2015-10-01T18:32:15.131Z - debug: closed write stream, unzipping now
2015-10-01T18:32:15.136Z - error: ERR unzipping:
2015-10-01T18:32:15.136Z - error:  0=32, 1=32, 2=69, 3=110, 4=100, 5=45, 6=111, 7=102, 8=45, 9=99, 10=101, 11=110, 12=116, 13=114, 14=97, 15=108, 16=45, 17=100, 18=105, 19=114, 20=101, 21=99, 22=116, 23=111, 24=114, 25=121, 26=32, 27=115, 28=105, 29=103, 30=110, 31=97, 32=116, 33=117, 34=114, 35=101, 36=32, 37=110, 38=111, 39=116, 40=32, 41=102, 42=111, 43=117, 44=110, 45=100, 46=46, 47=32, 48=32, 49=69, 50=105, 51=116, 52=104, 53=101, 54=114, 55=32, 56=116, 57=104, 58=105, 59=115, 60=32, 61=102, 62=105, 63=108, 64=101, 65=32, 66=105, 67=115, 68=32, 69=110, 70=111, 71=116, 72=10, 73=32, 74=32, 75=97, 76=32, 77=122, 78=105, 79=112, 80=102, 81=105, 82=108, 83=101, 84=44, 85=32, 86=111, 87=114, 88=32, 89=105, 90=116, 91=32, 92=99, 93=111, 94=110, 95=115, 96=116, 97=105, 98=116, 99=117, 100=116, 101=101, 102=115, 103=32, 104=111, 105=110, 106=101, 107=32, 108=100, 109=105, 110=115, 111=107, 112=32, 113=111, 114=102, 115=32, 116=97, 117=32, 118=109, 119=117, 120=108, 121=116, 122=105, 123=45, 124=112, 125=97, 126=114, 127=116, 128=32, 129=97, 130=114, 131=99, 132=104, 133=105, 134=118, 135=101, 136=46, 137=32, 138=32, 139=73, 140=110, 141=32, 142=116, 143=104, 144=101, 145=10, 146=32, 147=32, 148=108, 149=97, 150=116, 151=116, 152=101, 153=114, 154=32, 155=99, 156=97, 157=115, 158=101, 159=32, 160=116, 161=104, 162=101, 163=32, 164=99, 165=101, 166=110, 167=116, 168=114, 169=97, 170=108, 171=32, 172=100, 173=105, 174=114, 175=101, 176=99, 177=116, 178=111, 179=114, 180=121, 181=32, 182=97, 183=110, 184=100, 185=32, 186=122, 187=105, 188=112, 189=102, 190=105, 191=108, 192=101, 193=32, 194=99, 195=111, 196=109, 197=109, 198=101, 199=110, 200=116, 201=32, 202=119, 203=105, 204=108, 205=108, 206=32, 207=98, 208=101, 209=32, 210=102, 211=111, 212=117, 213=110, 214=100, 215=32, 216=111, 217=110, 218=10, 219=32, 220=32, 221=116, 222=104, 223=101, 224=32, 225=108, 226=97, 227=115, 228=116, 229=32, 230=100, 231=105, 232=115, 233=107, 234=40, 235=115, 236=41, 237=32, 238=111, 239=102, 240=32, 241=116, 242=104, 243=105, 244=115, 245=32, 246=97, 247=114, 248=99, 249=104, 250=105, 251=118, 252=101, 253=46, 254=10, 255=117, 256=110, 257=122, 258=105, 259=112, 260=58, 261=32, 262=32, 263=99, 264=97, 265=110, 266=110, 267=111, 268=116, 269=32, 270=102, 271=105, 272=110, 273=100, 274=32, 275=122, 276=105, 277=112, 278=102, 279=105, 280=108, 281=101, 282=32, 283=100, 284=105, 285=114, 286=101, 287=99, 288=116, 289=111, 290=114, 291=121, 292=32, 293=105, 294=110, 295=32, 296=111, 297=110, 298=101, 299=32, 300=111, 301=102, 302=32, 303=47, 304=85, 305=115, 306=101, 307=114, 308=115, 309=47, 310=108, 311=97, 312=99, 313=111, 314=108, 315=108, 316=105, 317=101, 318=114, 319=47, 320=68, 321=111, 322=99, 323=117, 324=109, 325=101, 326=110, 327=116, 328=115, 329=47, 330=109, 331=97, 332=118, 333=101, 334=110, 335=115, 336=109, 337=97, 338=116, 339=101, 340=47, 341=104, 342=101, 343=114, 344=102, 345=102, 346=106, 347=111, 348=110, 349=101, 350=115, 351=99, 352=111, 353=108, 354=108, 355=101, 356=103, 357=101, 358=47, 359=116, 360=109, 361=112, 362=46, 363=122, 364=105, 365=112, 366=32, 367=111, 368=114, 369=10, 370=32, 371=32, 372=32, 373=32, 374=32, 375=32, 376=32, 377=32, 378=47, 379=85, 380=115, 381=101, 382=114, 383=115, 384=47, 385=108, 386=97, 387=99, 388=111, 389=108, 390=108, 391=105, 392=101, 393=114, 394=47, 395=68, 396=111, 397=99, 398=117, 399=109, 400=101, 401=110, 402=116, 403=115, 404=47, 405=109, 406=97, 407=118, 408=101, 409=110, 410=115, 411=109, 412=97, 413=116, 414=101, 415=47, 416=104, 417=101, 418=114, 419=102, 420=102, 421=106, 422=111, 423=110, 424=101, 425=115, 426=99, 427=111, 428=108, 429=108, 430=101, 431=103, 432=101, 433=47, 434=116, 435=109, 436=112, 437=46, 438=122, 439=105, 440=112, 441=46, 442=122, 443=105, 444=112, 445=44, 446=32, 447=97, 448=110, 449=100, 450=32, 451=99, 452=97, 453=110, 454=110, 455=111, 456=116, 457=32, 458=102, 459=105, 460=110, 461=100, 462=32, 463=47, 464=85, 465=115, 466=101, 467=114, 468=115, 469=47, 470=108, 471=97, 472=99, 473=111, 474=108, 475=108, 476=105, 477=101, 478=114, 479=47, 480=68, 481=111, 482=99, 483=117, 484=109, 485=101, 486=110, 487=116, 488=115, 489=47, 490=109, 491=97, 492=118, 493=101, 494=110, 495=115, 496=109, 497=97, 498=116, 499=101, 500=47, 501=104, 502=101, 503=114, 504=102, 505=102, 506=106, 507=111, 508=110, 509=101, 510=115, 511=99, 512=111, 513=108, 514=108, 515=101, 516=103, 517=101, 518=47, 519=116, 520=109, 521=112, 522=46, 523=122, 524=105, 525=112, 526=46, 527=90, 528=73, 529=80, 530=44, 531=32, 532=112, 533=101, 534=114, 535=105, 536=111, 537=100, 538=46, 539=10, length=540, parent=undefined
2015-10-01T18:32:15.145Z - error: error retrieving and writing to disk
2015-10-01T18:32:15.145Z - error: Could not extract and write stream to file system.
2015-10-01T18:32:15.145Z - error: Could not retrieve and write project to file system: Could not extract and write stream to file system.
2015-10-01T18:32:15.146Z - error: Error: Could not extract and write stream to file system.
    at /Applications/MavensMate.app/Contents/Resources/app/node_modules/mavensmate/lib/mavensmate/util.js:354:35
    at tryCatcher (/Applications/MavensMate.app/Contents/Resources/app/node_modules/bluebird/js/main/util.js:26:23)
    at Promise._settlePromiseFromHandler (/Applications/MavensMate.app/Contents/Resources/app/node_modules/bluebird/js/main/promise.js:503:31)
    at Promise._settlePromiseAt (/Applications/MavensMate.app/Contents/Resources/app/node_modules/bluebird/js/main/promise.js:577:18)
    at Promise._settlePromises (/Applications/MavensMate.app/Contents/Resources/app/node_modules/bluebird/js/main/promise.js:693:14)
    at Async._drainQueue (/Applications/MavensMate.app/Contents/Resources/app/node_modules/bluebird/js/main/async.js:123:16)
    at Async._drainQueues (/Applications/MavensMate.app/Contents/Resources/app/node_modules/bluebird/js/main/async.js:133:10)
    at Immediate.Async.drainQueues [as _onImmediate] (/Applications/MavensMate.app/Contents/Resources/app/node_modules/bluebird/js/main/async.js:15:14)
    at processImmediate [as _immediateCallback] (timers.js:371:17)
2015-10-01T18:32:15.146Z - error: Could not initiate new Project instance: Could not extract and write stream to file system.
```

MavensMate.app is on version 0.0.7-beta4

**Comment by joeferraro** [comment]
@lacollier can you describe the types of metadata you're trying to pull down? Does this happen for any project?

**Comment by lacollier** [comment]
I upgraded MavensMate.app to 0.0.8, as well as upgraded the plugin to v5.0.4.

I tried another project and it worked.

Are there specific types of metadata that will break the new project process?  

Thank you.

**Comment by joeferraro** [comment]
@lacollier should not. it'd be helpful to understand what your project contained. if i had to guess, i'd say you were probably downloading static resource(s) as part of your project?

**Comment by lacollier** [comment]
Here is my package.xml file, hopefully that will get you some useful information:

```
<?xml version="1.0" encoding="UTF-8"?>
<Package xmlns="http://soap.sforce.com/2006/04/metadata">
    <types>
        <members>*</members>
        <name>AnalyticSnapshot</name>
    </types>
    <types>
        <members>*</members>
        <name>ApexClass</name>
    </types>
    <types>
        <members>*</members>
        <name>ApexComponent</name>
    </types>
    <types>
        <members>*</members>
        <name>ApexPage</name>
    </types>
    <types>
        <members>*</members>
        <name>ApexTrigger</name>
    </types>
    <types>
        <members>*</members>
        <name>ApexTriggerCoupling</name>
    </types>
    <types>
        <members>*</members>
        <name>ApprovalProcess</name>
    </types>
    <types>
        <members>*</members>
        <name>AssignmentRule</name>
    </types>
    <types>
        <members>*</members>
        <name>AssignmentRules</name>
    </types>
    <types>
        <members>*</members>
        <name>AuthProvider</name>
    </types>
    <types>
        <members>*</members>
        <name>AutoResponseRule</name>
    </types>
    <types>
        <members>*</members>
        <name>AutoResponseRules</name>
    </types>
    <types>
        <members>*</members>
        <name>BusinessProcess</name>
    </types>
    <types>
        <members>*</members>
        <name>CallCenter</name>
    </types>
    <types>
        <members>*</members>
        <name>Community</name>
    </types>
    <types>
        <members>*</members>
        <name>CompactLayout</name>
    </types>
    <types>
        <members>*</members>
        <name>ConnectedApp</name>
    </types>
    <types>
        <members>*</members>
        <name>CustomApplication</name>
    </types>
    <types>
        <members>*</members>
        <name>CustomApplicationComponent</name>
    </types>
    <types>
        <members>*</members>
        <name>CustomField</name>
    </types>
    <types>
        <members>*</members>
        <name>CustomLabel</name>
    </types>
    <types>
        <members>*</members>
        <name>CustomLabels</name>
    </types>
    <types>
        <members>*</members>
        <members>Account</members>
        <members>AccountContactRole</members>
        <members>Activity</members>
        <members>Asset</members>
        <members>Campaign</members>
        <members>CampaignMember</members>
        <members>Case</members>
        <members>CaseContactRole</members>
        <members>Contact</members>
        <members>ContentVersion</members>
        <members>Contract</members>
        <members>ContractContactRole</members>
        <members>Event</members>
        <members>Lead</members>
        <members>Opportunity</members>
        <members>OpportunityContactRole</members>
        <members>OpportunityLineItem</members>
        <members>Order</members>
        <members>OrderItem</members>
        <members>PartnerRole</members>
        <members>Pricebook2</members>
        <members>PricebookEntry</members>
        <members>Product2</members>
        <members>Site</members>
        <members>Solution</members>
        <members>Task</members>
        <members>User</members>
        <name>CustomObject</name>
    </types>
    <types>
        <members>*</members>
        <name>CustomPageWebLink</name>
    </types>
    <types>
        <members>*</members>
        <name>CustomSite</name>
    </types>
    <types>
        <members>*</members>
        <name>CustomTab</name>
    </types>
    <types>
        <members>*</members>
        <name>Dashboard</name>
    </types>
    <types>
        <members>*</members>
        <name>DataCategoryGroup</name>
    </types>
    <types>
        <members>*</members>
        <name>Document</name>
    </types>
    <types>
        <members>*</members>
        <name>EmailTemplate</name>
    </types>
    <types>
        <members>*</members>
        <name>EscalationRule</name>
    </types>
    <types>
        <members>*</members>
        <name>EscalationRules</name>
    </types>
    <types>
        <members>*</members>
        <name>FieldSet</name>
    </types>
    <types>
        <members>*</members>
        <name>FlexiPage</name>
    </types>
    <types>
        <members>*</members>
        <name>Group</name>
    </types>
    <types>
        <members>*</members>
        <name>HomePageComponent</name>
    </types>
    <types>
        <members>*</members>
        <name>HomePageLayout</name>
    </types>
    <types>
        <members>*</members>
        <name>Layout</name>
    </types>
    <types>
        <members>*</members>
        <name>Letterhead</name>
    </types>
    <types>
        <members>*</members>
        <name>ListView</name>
    </types>
    <types>
        <members>*</members>
        <name>NamedFilter</name>
    </types>
    <types>
        <members>*</members>
        <name>PermissionSet</name>
    </types>
    <types>
        <members>*</members>
        <name>Profile</name>
    </types>
    <types>
        <members>*</members>
        <name>Queue</name>
    </types>
    <types>
        <members>*</members>
        <name>QuickAction</name>
    </types>
    <types>
        <members>*</members>
        <name>RecordType</name>
    </types>
    <types>
        <members>*</members>
        <name>RemoteSiteSetting</name>
    </types>
    <types>
        <members>*</members>
        <name>Report</name>
    </types>
    <types>
        <members>*</members>
        <name>ReportType</name>
    </types>
    <types>
        <members>*</members>
        <name>Role</name>
    </types>
    <types>
        <members>*</members>
        <name>SamlSsoConfig</name>
    </types>
    <types>
        <members>*</members>
        <name>Scontrol</name>
    </types>
    <types>
        <members>*</members>
        <name>SharingReason</name>
    </types>
    <types>
        <members>*</members>
        <name>StaticResource</name>
    </types>
    <types>
        <members>*</members>
        <name>SynonymDictionary</name>
    </types>
    <types>
        <members>*</members>
        <name>ValidationRule</name>
    </types>
    <types>
        <members>*</members>
        <name>WebLink</name>
    </types>
    <types>
        <members>*</members>
        <name>Workflow</name>
    </types>
    <types>
        <members>*</members>
        <name>WorkflowAlert</name>
    </types>
    <types>
        <members>*</members>
        <name>WorkflowApex</name>
    </types>
    <types>
        <members>*</members>
        <name>WorkflowChatterPost</name>
    </types>
    <types>
        <members>*</members>
        <name>WorkflowFieldUpdate</name>
    </types>
    <types>
        <members>*</members>
        <name>WorkflowKnowledgePublish</name>
    </types>
    <types>
        <members>*</members>
        <name>WorkflowOutboundMessage</name>
    </types>
    <types>
        <members>*</members>
        <name>WorkflowQuickCreate</name>
    </types>
    <types>
        <members>*</members>
        <name>WorkflowRule</name>
    </types>
    <types>
        <members>*</members>
        <name>WorkflowSend</name>
    </types>
    <types>
        <members>*</members>
        <name>WorkflowTask</name>
    </types>
    <types>
        <members>*</members>
        <name>Settings</name>
    </types>
    <version>34.0</version>
</Package>
```

**Comment by eofferma** [comment]
I have this error too on MavensMate-app 0.09 and I suppose that is related with the StaticResources metadatatype. Is maybe related with the StaticResources that are not ZIP files?

**Comment by joeferraro** [comment]
@eofferma platform & architecture?

**Comment by Yerrak** [comment]
_Leaving this here as an additional piece of information for potential future reference :)_

This happened to me as well today. I Created a project from scratch and selected all metadata.
For me, the `StaticResources` (as suggested as the 'corrupt' metadata entry by @eofferma) worked well and I was able to narrow it down to `Document` metadata in my case.

Wild guess from my side here (as my current org has a large amount of data for `Document` and `StaticResource` might also be quite extensive): the folder-/file-hierarchy below the metadata entry or the total size (number?) of elements contained may be a reason for this error?

Also the error only occured when I tried to _update_ the project. Not on initial creation.

 
_My current setup:_
* Sublime Text Build 3114
* MavensMate-app-v0.0.10-osx-x64
* MavensMate for Sublime Text v7.0.0-beta.3
* OS X 10.11.5
